dotCMS/src/main/java/com/dotcms/rest/api/v1/storage/FileStorageResource.java

```diff
@@ -0,0 +1,392 @@
+import com.dotcms.util.ConversionUtils;
+import com.dotcms.util.DotPreconditions;
+import com.dotcms.uuid.shorty.ShortyId;
+import com.dotcms.workflow.form.FireActionForm;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.web.WebAPILocator;
+import com.dotmarketing.exception.DoesNotExistException;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.business.ContentletAPI;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.PageMode;
+import com.dotmarketing.util.StringUtils;
+import com.dotmarketing.util.UtilMethods;
+import com.google.common.collect.ImmutableMap;
+import com.liferay.portal.language.LanguageUtil;
+import com.liferay.portal.model.User;
+import io.vavr.Tuple2;
+import io.vavr.control.Try;
+import org.glassfish.jersey.media.multipart.FormDataMultiPart;
+import org.glassfish.jersey.server.JSONP;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Supplier;
+
+@Path("/v1/storage")
+public class FileStorageResource {
+
+    private final WebResource    webResource;
+    private final FileStorageAPI fileStorageAPI;
+    private final MultiPartUtils multiPartUtils;
+    private final ContentletAPI contentletAPI;
+    private final ContentletMetadataAPI contentletMetadataAPI;
+
+    @VisibleForTesting
+    protected FileStorageResource(final FileStorageAPI fileStorageAPI,
+                                  final WebResource webResource,
+                                  final MultiPartUtils multiPartUtils,
+                                  final ContentletAPI contentletAPI,
+                                  final ContentletMetadataAPI contentletMetadataAPI) {
+
+        this.fileStorageAPI = fileStorageAPI;
+        this.webResource    = webResource;
+        this.multiPartUtils = multiPartUtils;
+        this.contentletAPI  = contentletAPI;
+        this.contentletMetadataAPI = contentletMetadataAPI;
+    }
+
+    public FileStorageResource() {
+        this(APILocator.getFileStorageAPI(), new WebResource(),
+                new MultiPartUtils(), APILocator.getContentletAPI(), APILocator.getContentletMetadataAPI());
+    }
+
+    @PUT
+    @Path("/metadata/basic")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String literal "application/javascript" appears 5 times in this file; the first occurrence is on line 95](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5547873)